src/pages/workspace/accounting/intacct/SageIntacctEntityPage.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import RadioListItem from '@components/SelectionList/RadioListItem';
+import type {ListItem} from '@components/SelectionList/types';
+import SelectionScreen from '@components/SelectionScreen';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {clearSageIntacctErrorField, updateSageIntacctEntity} from '@libs/actions/connections/SageIntacct';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import withPolicy from '@pages/workspace/withPolicy';
+import type {WithPolicyProps} from '@pages/workspace/withPolicy';
+import CONST from '@src/CONST';
+
+function SageIntacctEntityPage({policy}: WithPolicyProps) {
+    const styles = useThemeStyles();
+    const config = policy?.connections?.intacct?.config;
+    const entityID = config?.entity ?? '';
+
+    const policyID = policy?.id ?? '-1';
+
+    const sections =
+        policy?.connections?.intacct?.data?.entities.map((entity) => ({
+            text: entity.name,
+            value: entity.name,
+            keyForList: entity.id,
+            isSelected: entity.id === entityID,
+        })) ?? [];
+
+    const saveSelection = ({text, keyForList}: ListItem) => {
+        if (!keyForList) {
+            return;
+        }
+
+        updateSageIntacctEntity(policyID, text ?? '');
```

Comment:
We updated the entity as text here but we're using entity id to check the selected option, it will cause this bug:

https://github.com/user-attachments/assets/c5e012c4-a7f1-454c-93b8-18b63c0852f3